server/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -1060,6 +1073,17 @@ public <T> List<T> readList(Writeable.Reader<T> reader) throws IOException {
         return list;
     }
 
+    /**
+     * Reads a list of strings. The list is expected to have been written using {@link StreamOutput#writeStringCollection(Collection)}.
+     * If the returned list contains any entries it will be mutable. If it is empty it might be immutable.
+     *
+     * @return the list of strings
+     * @throws IOException if an I/O exception occurs reading the list
+     */
+    public List<String> readStringList() throws IOException {
```

Comment:
this is not part of backported commit but since we don't have such method I added it as well